app-modules/student-data-model/src/Filament/Resources/StudentResource/Pages/EditStudent.php

```diff
@@ -191,6 +199,32 @@ public function form(Form $form): Form
                                     ->label('Can receive SMS messages')
                                     ->columnSpanFull()
                                     ->default(true),
+                                Checkbox::make('sms_opt_out_phone_number')
+                                    ->label('SMS Opt Out')
+                                    ->columnSpanFull()
+                                    ->default(false)
+                                    ->dehydrated(false)
+                                    ->live()
+                                    ->afterStateHydrated(function (Component $component): void {
+                                        $phoneNumber = $component->getContainer()->getRecord();
+
+                                        if ($phoneNumber instanceof StudentPhoneNumber && $phoneNumber->exists) {
+                                            $isOptedOut = app(SmsOptOutService::class)->isStudentPhoneNumberOptedOut($phoneNumber);
+                                            $component->state($isOptedOut);
+                                        }
+                                    })
+                                    ->afterStateUpdated(function (mixed $state, Component $component): void {
+                                        $record = $component->getContainer()->getRecord();
+
+                                        if ($record instanceof StudentPhoneNumber && $record->exists) {
+                                            $key = $record->id;
+                                        } else {
+                                            $key = $component->getContainer()->getStatePath();
+                                        }
+
+                                        $this->smsOptOutStates[$key] = (bool) $state;
+                                    })
```

Comment:
Instead of doing this logic here, I would probably use the `mutateRelationshipData` hooks before fill and before save. https://filamentphp.com/docs/3.x/forms/fields/repeater#mutating-related-item-data-before-filling-the-field

Before fill, we can set the value to be true or false based on if the number is in the opted out table.

Before save, we can save the number to the table, or delete it from the table if it has changed.

Then we can remove the `smsOptOutStates` property for tracking this, since the information is already available in the `$data` of each hook method.